kogito-codegen/src/main/java/org/kie/kogito/codegen/decision/DecisionContainerGenerator.java

```
@@ -71,6 +73,14 @@ public ClassOrInterfaceDeclaration classDeclaration() {
             }
         }
 
+        if (addonsConfig.useMonitoring()) {
+            typeDeclaration.getMethodsByName("getDecisionModel").stream().findFirst()
+                    .flatMap(md -> md.findFirst(ReturnStmt.class))
+                    .ifPresent(rs -> rs.getExpression().ifPresent(rsExp ->
+                            rs.setExpression(newObject("org.kie.kogito.monitoring.decision.MonitoringDecisionModel", rsExp))
+                    ));
```

In this way you are hiding any possible error (i.e. template changed). What about preserve the optional and then if empty throws an error?

---

You're right. I originally used `ifPresentOrElse` but it's a Java 9+ API and we must stick to Java 8 for `kogito-runtimes`.
I changed now anyway to throw an exception if the template is changed. :+1: